zebra-network/src/config.rs

```diff
@@ -37,19 +32,53 @@ pub struct Config {
 }
 
 impl Config {
-    fn parse_peers<S: ToSocketAddrs>(peers: HashSet<S>) -> HashSet<SocketAddr> {
-        peers
-            .iter()
-            .flat_map(|s| s.to_socket_addrs())
-            .flatten()
-            .collect()
+    async fn parse_peers(peers: HashSet<String>) -> HashSet<SocketAddr> {
```

Comment:
Let's document this function as well:
```suggestion
    /// Concurrently resolves `peers` into zero or more IP addresses, with a timeout
    /// of a few seconds on each DNS request.
    ///
    /// If DNS resolution fails or times out for all peers, returns an empty list.
    async fn parse_peers(peers: HashSet<String>) -> HashSet<SocketAddr> {
```